Makefile

```diff
@@ -1,4 +1,4 @@
-CFLAGS := $(CFLAGS) -O2 -ffast-math -fstrict-aliasing -march=native
+CFLAGS := $(CFLAGS) -O2 -ffast-math -fstrict-aliasing -march=native -x cuda
```

Comment:
I think we can't do this always, otherwise the tests won't work on GCC at least? I think a separate build test like you have below is good though.

Reply:
Didn't mean to push that. Makefile is nicer now.